packages/core/src/utils/bigNumberToLocaleString.ts

```diff
@@ -0,0 +1,38 @@
+import BigNumber from 'bignumber.js';
+
+const complexNumber = 1234567.0123456789;
+
+export default function bigNumberToLocaleString(value: BigNumber, locale?: string): string {
+    const formatter = Intl.NumberFormat(locale);
+    if (!formatter) {
+      throw new Error(`Formater for ${locale} is not supported`);
+    }
+
+    const decimalFormatter = new Intl.NumberFormat(locale, { 
+      maximumFractionDigits: 12,
+    });
+    if (!decimalFormatter) {
+      throw new Error(`Decimal formater for ${locale} is not supported`);
+    }
+
+    const parts = decimalFormatter.formatToParts(complexNumber);
+
+    const decimalPart = parts.find(part => part.type === 'decimal');
+    const groupPart = parts.find(part => part.type === 'group');
+
+    const reversedParts = parts.slice().reverse();
+    const integerPart = reversedParts.find(part => part.type === 'integer');
+
+    const format = {
```

Comment:
Just curious if this format also handles negative values properly.

Reply:
It is working fine with negative values